medcat-service/medcat_service/nlp_processor/medcat_processor.py

```diff
@@ -84,7 +92,34 @@ def __init__(self):
         self.cat = self._create_cat()
         self.cat.train = os.getenv("APP_TRAINING_MODE", False)
 
-        self.log.info("MedCAT processor is ready")
+        self._is_ready_flag = self._check_medcat_readiness()
+
+    def _check_medcat_readiness(self) -> bool:
+        readiness_text = "MedCAT is ready and can get_entities"
+        try:
+            result = self.cat.get_entities(readiness_text)
+            self.log.debug("Result of readiness check is" + str(result))
+            self.log.info("MedCAT processor is ready")
+            return True
+        except Exception as e:
+            self.log.error(f"MedCAT processor is not ready. Failed the readiness check: {e}")
```

Comment:
If we pass the error to the logger through `exc_info=e` keyword argument, we can log the full stack trace instead of just the message (which the current implementation would show).